benchmarks/libmagic/README.md

```diff
@@ -0,0 +1,236 @@
+# Benchmarking libmagic with minimal magic files
+
+To assess the performance improvement gained by using a minimal magic file, we created the benchmark
+`main.c`. This benchmark measures the time needed to find the MIME type of a file with
+a given compiled magic file.
+
+## Usage
+
+The program can be compiled with the following command:
+
+```bash
+gcc main.c -lmagic -o bench
+```
+
+This assumes that you have `libmagic` installed on you computer.
+
+Once the the program has been compiled, you can run it with the (mandatory) arguments specifying the path to the
+compiled magic file, the path to the directory conatining the test files and the number of iterations with the following command:
+
+```bash
+./bench <path to magic file> <path to test files dir> <number of iterations>
+```
+
+This gives you the mean, the variance and the standard deviation of the measurements (per file). It also outputs a file containing those values in a markdown table format. 
+
+## Magic files
+
+In `magic_files/` we provide 3 tests magic files:
+- magic.mgc: the default compiled magic file for the version 5.39 of libmagic
+- middle.mgc: compiled magic file obtained running the CLI tool with the option `-a` (and the magic files for the version 5.39 of libmagic). This magic file contains 1035 tests.
+- small.mgc:  compiled magic file obtained running the CLI tool with the option `--mime-types "biosig/atf,text/x-nawk,application/x-sega-cd-rom,application/x-dbf,application/vnd.stardivision.math,image/heic,video/x-jng,image/x-os2-graphics,application/pdf,application/javascript,application/x-garmin-typ,biosig/ced-smr,application/x-iso9660-image"` (and the magic files for the version 5.39 of libmagic). This magic file contains 35 tests.
+
+Feel free to test it with your own magic file.
+
+## Results
+
+We gathered the results of the benchmarks in th following tables. Those results were obtained with `5000` iterations on a `13-inch 2017 MacBook Pro` with a `3.1 GHz Dual-Core Intel Core i5` and `16 Gb` of RAM.
+
+### Comparison
+
+We focus here on the file `application-pdf.pdf` as it is detected by all compiled magic files. The results are similar in this scenario for the other files. There average time of the benchmarks for each magic file is:
```

Comment:
Focus is a bit confusing I find as you show the results for many more mime-types. Maybe say that the comparison of the average time it takes to detect application/pdf is shown for the three different sizes of magic DBs.